Hi, I would like some help with the error message I am getting in the checks. I created the .rst file and ran make_docs which created the build folder in which I confirmed the .HTML file for the function is created. The .html is not detected when I try to commit it. Clearly I'm doing something wrong in the docs section but I'm not sure where the problem is.

This is the error that Azure pipeline gives:

```Traceback (most recent call last):
  File "scripts/check-autodoc.py", line 73, in <module>
    f"The following functions have not yet been added: {missing_funcs}"
Exception: The following functions have not yet been added: {'groupby_topk'}
```

`

--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -1,5 +1,6 @@
 new version (on deck)
 =====================
+- [ENH] Updated groupby_agg function to account for null entries in the ``by`` argument. @samukweku
 - [ENH] Added function ``groupby_topk`` to janitor functions @mphirke
 
 

--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -1,5 +1,6 @@
 new version (on deck)
 =====================
+- [ENH] Added function ``groupby_topk`` to janitor functions @mphirke
 
 
 v0.20.8
@@ -21,7 +22,6 @@ v0.20.8
 - [ENH] Added ``fill_direction`` function for forward/backward fills on missing values
   for selected columns in a dataframe. @samukweku
 - [ENH] Simpler logic and less lines of code for expand_grid function @samukweku
-- [ENH] Added function ``groupby_topk`` to janitor functions @mphirke
 
 
 v0.20.7

--- a/docs/reference/general_functions.rst
+++ b/docs/reference/general_functions.rst
@@ -88,4 +88,5 @@ Other
     move
     toset
     unionize_dataframe_categories
+    groupby_topk
     

--- a/janitor/functions.py
+++ b/janitor/functions.py
@@ -4256,11 +4256,16 @@ def groupby_topk(
     sort_values_kwargs: Dict = None,
 ) -> pd.DataFrame:
     """
-    Return a dataframe that has the top `k` values grouped by
-    `groupby_column_name` and sorted by `sort_column_name`. Additional
-    parameters to the sorting (such as ascending=True) can be passed using
-    `sort_values_kwargs`. List of all sort_values() parameters can be found at
-    https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.sort_values.html
+    Return top `k` rows from a groupby of a set of columns.
+
+    Returns a dataframe that has the top `k` values grouped by `groupby_column_name`
+    and sorted by `sort_column_name`.
+    Additional parameters to the sorting (such as ascending=True)
+    can be passed using `sort_values_kwargs`.
+
+    List of all sort_values() parameters can be found here_.
+
+    .. _here: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.sort_values.html
 
 
     .. code-block:: python
@@ -4334,11 +4339,9 @@ def groupby_topk(
         `groupby_column_name` column with each group sorted along the
         column `sort_column_name`.
     :raises: ValueError if `k` is less than 1.
-    :raises: ValueError if `groupb y_column_name` does not exist in dataframe.
-    :raises: ValueError if `sort_column_name` does not exist in dataframe.
     :raises: ValueError if `groupby_column_name` is same as `sort_column_name`.
     :raises: KeyError if `inplace:True` is present in `sort_values_kwargs`.
-    """
+    """  # noqa: E501
 
     # Convert the default sort_values_kwargs from None to empty Dict
     sort_values_kwargs = sort_values_kwargs or {}
@@ -4347,18 +4350,6 @@ def groupby_topk(
     if k < 1:
         raise ValueError(
             "Numbers of rows per group to be returned must be greater than 0."
-        )
-
-    # Check if groupby_column_name exists in dataframe.
-    if groupby_column_name not in df.columns:
-        raise ValueError(
-            f"Column {groupby_column_name} does not exist in the dataframe."
-        )
-
-    # Check if sort_column_name exists in dataframe.
-    if sort_column_name not in df.columns:
-        raise ValueError(
-            f"Column {sort_column_name} does not exist in the dataframe."
         )
 
     # Check if group by column and sort column are unique.
@@ -4370,7 +4361,7 @@ def groupby_topk(
         "inplace" in sort_values_kwargs.keys()
         and sort_values_kwargs["inplace"]
     ):
-        raise KeyError("Cannot use inplace : True in sort_values_kwargs.")
+        raise KeyError("Cannot use `inplace=True` in `sort_values_kwargs`.")
 
     return df.groupby(groupby_column_name).apply(
         lambda d: d.sort_values(sort_column_name, **sort_values_kwargs).head(k)

--- a/janitor/functions.py
+++ b/janitor/functions.py
@@ -4339,7 +4339,6 @@ def groupby_topk(
         `groupby_column_name` column with each group sorted along the
         column `sort_column_name`.
     :raises: ValueError if `k` is less than 1.
-    :raises: ValueError if `groupby_column_name` is same as `sort_column_name`.
     :raises: KeyError if `inplace:True` is present in `sort_values_kwargs`.
     """  # noqa: E501
 
@@ -4351,10 +4350,6 @@ def groupby_topk(
         raise ValueError(
             "Numbers of rows per group to be returned must be greater than 0."
         )
-
-    # Check if group by column and sort column are unique.
-    if sort_column_name == groupby_column_name:
-        raise ValueError("Group by column and sort column cannot be same.")
 
     # Check if inplace:True in sort values kwargs because it returns None
     if (

--- a/scripts/check-autodoc.py
+++ b/scripts/check-autodoc.py
@@ -70,5 +70,7 @@ missing_funcs = set(function_names).difference(doc_functions)
 
 if len(missing_funcs) > 0:
     raise Exception(
-        f"The following functions have not yet been added: {missing_funcs}"
+        f'''The following functions have not yet been added: {missing_funcs}.
+         Please add the function names to docs/reference/general_functions.rst
+         under an appropriate section.'''
     )

--- a/scripts/check-autodoc.py
+++ b/scripts/check-autodoc.py
@@ -70,7 +70,7 @@ missing_funcs = set(function_names).difference(doc_functions)
 
 if len(missing_funcs) > 0:
     raise Exception(
-        f'''The following functions have not yet been added: {missing_funcs}.
+        f"""The following functions have not yet been added: {missing_funcs}.
          Please add the function names to docs/reference/general_functions.rst
-         under an appropriate section.'''
+         under an appropriate section."""
     )

--- a/tests/functions/test_groupby_topk.py
+++ b/tests/functions/test_groupby_topk.py
@@ -55,12 +55,6 @@ def test_negative_k(df):
         df.groupby_topk("result", "age", 0)
 
 
-def test_same_sort_groupby_columns(df):
-    """Raises Value Error if columns to group by and sort along are same."""
-    with pytest.raises(ValueError):
-        df.groupby_topk("result", "result", 2)
-
-
 def test_inplace(df):
     """Raise Key Error if inplace is True in sort_values_kwargs"""
     with pytest.raises(KeyError):

--- a/tests/functions/test_groupby_topk.py
+++ b/tests/functions/test_groupby_topk.py
@@ -37,13 +37,13 @@ def test_descending_groupby_k_3(df):
 
 def test_wrong_groupby_column_name(df):
     """Raise Value Error if wrong groupby column name is provided."""
-    with pytest.raises(ValueError):
+    with pytest.raises(KeyError):
         df.groupby_topk("RESULT", "age", 3)
 
 
 def test_wrong_sort_column_name(df):
     """Raise Value Error if wrong sort column name is provided."""
-    with pytest.raises(ValueError):
+    with pytest.raises(KeyError):
         df.groupby_topk("result", "Age", 3)
 
 

--- a/tests/functions/test_groupby_topk.py
+++ b/tests/functions/test_groupby_topk.py
@@ -19,7 +19,6 @@ def df():
 
 def test_ascending_groupby_k_2(df):
     """Test ascending group by, k=2"""
-    expected = df.copy()
     expected = df.groupby("result").apply(
         lambda d: d.sort_values("age").head(2)
     )
@@ -28,7 +27,6 @@ def test_ascending_groupby_k_2(df):
 
 def test_descending_groupby_k_3(df):
     """Test descending group by, k=3"""
-    expected = df.copy()
     expected = df.groupby("result").apply(
         lambda d: d.sort_values("age", ascending=False).head(3)
     )

